esmvaltool/diag_scripts/perfmetrics/portrait_plot.py

```diff
@@ -0,0 +1,476 @@
+   bbox_inches="tight", which can change the resulting aspect ratio.
+   By default [5, 3].
+dpi: int, optional
+    Dots per inch for the figure. By default 300.
+"""
+
+import itertools
+import logging
+
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import numpy as np
+import xarray as xr
+from matplotlib import patches
+from mpl_toolkits.axes_grid1 import ImageGrid
+
+from esmvaltool.diag_scripts.shared import (
+    get_plot_filename,
+    group_metadata,
+    run_diagnostic,
+    select_metadata,
+)
+
+log = logging.getLogger(__name__)
+
+
+def unify_limits(grid):
+    """Set same limits for all subplots."""
+    vmin, vmax = np.inf, -np.inf
+    images = [ax.get_images()[0] for ax in grid]
+    for img in images:
+        vmin = min(vmin, img.get_clim()[0])
+        vmax = max(vmax, img.get_clim()[1])
+    for img in images:
+        img.set_clim(vmin, vmax)
+
+
+def plot_matrix(data, row_labels, col_labels, axe, plot_kwargs):
+    """Create an image for given data."""
+    img = axe.imshow(data, **plot_kwargs)
+    # Show all ticks and label them with the respective list entries.
+    axe.set_xticks(np.arange(data.shape[1]), labels=col_labels)
+    axe.set_yticks(np.arange(data.shape[0]), labels=row_labels)
+    # Rotate the tick labels and set their alignment.
+    plt.setp(
+        axe.get_xticklabels(),
+        rotation=90,
+        ha="right",
+        va="center",
+        rotation_mode="anchor",
+    )
+    # Turn spines off and create white grid.
+    # ax.spines[:].set_visible(False)
+    axe.set_xticks(np.arange(data.shape[1] + 1) - 0.5, minor=True)
+    axe.set_yticks(np.arange(data.shape[0] + 1) - 0.5, minor=True)
+    axe.grid(which="minor", color="black", linestyle="-", linewidth=1)
+    axe.tick_params(which="both", bottom=False, left=False)
+    return img
+
+
+def remove_reference(metas):
+    """Remove reference for metric from list of metadata.
+
+    NOTE: list() creates a copy with same references to allow removing in place
+    """
+    for meta in list(metas):
+        if meta.get("reference_for_metric", False):
+            metas.remove(meta)
+
+
+def add_split_none(metas):
+    """List of metadata with split=None if no split is given."""
+    for meta in metas:
+        if "split" not in meta:
+            meta["split"] = None
+
+
+def open_file(metadata, **selection):
+    """Try to find a single file for selection and return data.
+
+    If multiple files are found, raise an error. If no file is found,
+    return np.nan.
+    """
+    metas = select_metadata(metadata, **selection)
+    if len(metas) > 1:
+        raise ValueError(f"Multiple files found for {selection}")
+    if len(metas) < 1:
+        log.debug("No Metadata found for %s", selection)
+        return np.nan
+    log.warning("Metadata found for %s", selection)
+    das = xr.open_dataset(metas[0]["filename"])
+    varname = list(das.data_vars.keys())[0]
+    return das[varname].values.item()
```

Comment:
Would be great to raise a nicer error here if the data is non-scalar, at the moment this gives `ValueError: can only convert an array of size 1 to a Python scalar`.

Reply:
Sounds good!